### PR TITLE
Expose connects_to configuration for ActiveStorage models

### PR DESCRIPTION
**Pull request by npezza93**
### Detail

This allows people to set where ActiveStorage tables live without having to setup an initializer in their app. This is the same setup that Solid* adopts.

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Unrelated changes should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [ ] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

File: activestorage/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 ## Rails 8.0.0.beta1 (September 26, 2024) ##
 
+*   Expose `connects_to` configuration for models.
+
+    *Nick Pezza*
+
```
Comment (skipkayhil) [inline activestorage/CHANGELOG.md:6]: This should go above the `8.0.0.beta1` header since we can't change things that have already been released

Comment (npezza93) [reply activestorage/CHANGELOG.md:4]: Thanks, should be all fixed now

**Comment by rafaelfranca** [comment]
```ruby
ActiveSupport.on_load(:active_storage_record) do
  connects_to(some_config_I_want_to_use)
end
```

No need syntax sugar for that.

**Comment by zzak** [comment]
This is similar to #52569, which I thought was a step in the right direction but it sounds like yagni

**Pull request closed without merging**